### PR TITLE
Change Prebuilt OD For Polygons

### PR DESCRIPTION
**Pull request by marko-polo-cheno**

### CODE REVIEW

File: kolena/_experimental/object_detection/workflow.py [hunk 1]
```diff
@@ -50,35 +52,38 @@ class TestSample(Image):
...
-        object.__setattr__(self, "n_bboxes", len(self.bboxes))
+        object.__setattr__(self, "n_objects", len(self.objects))
 
 
 @dataclass(frozen=True)
 class Inference(BaseInference):
     """Inference type for the pre-built 2D Object Detection workflow."""
 
-    bboxes: List[ScoredLabeledBoundingBox]
+    objects: List[Union[ScoredLabeledBoundingBox, ScoredLabeledPolygon]]
```
Comment (ly-kolena) [inline kolena/_experimental/object_detection/workflow.py]: Would there be a list of mixed bboxes and polygons?

Comment (marko-polo-cheno) [reply kolena/_experimental/object_detection/workflow.py]: Not commonly seen but they can coexist, neither has more significance than the other.

File: kolena/_experimental/object_detection/workflow.py [hunk 2]
```diff
@@ -71,14 +75,15 @@ def __post_init__(self):
...
-    The inference [`ScoredLabeledBoundingBox`][kolena.workflow.annotation.ScoredLabeledBoundingBox]es associated with
-    an image.
+    The inference [`ScoredLabeledBoundingBox`][kolena.workflow.annotation.ScoredLabeledBoundingBox]es or
+    [`ScoredLabeledPolygon`][kolena.workflow.annotation.ScoredLabeledPolygon]s associated with an image.
     """
+
     ignored: bool = False
     """
-    Whether the image (and its associated inference `bboxes`) should be ignored in evaluating the results of the model.
+    Whether the image (and its associated inference `objects`) should be ignored in evaluating the results of the model.
```
Comment (gordonhart) [inline kolena/_experimental/object_detection/workflow.py]: Revert

File: kolena/_experimental/object_detection/workflow.py [hunk 3]
```diff
@@ -185,6 +190,7 @@ class TestCaseMetrics(MetricsTestCase):
 class TestSuiteMetrics(MetricsTestSuite):
     n_images: int
     mean_AP: float
+    threshold: Union[float, None] = None
```
Comment (gordonhart) [inline kolena/_experimental/object_detection/workflow.py]: Nit: prefer `Optional[float]`

**Review by gordonhart — APPROVED** [review]
I like this approach @consoull. Preemptively approving with two minor nits.

**Pull request merged**